File: backend/TransparenSee/src/main/java/gov/transparenSee/TransparenSee/models/Company.java
package gov.transparenSee.TransparenSee.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Document(collection = "companies")
public class Company {
    @Id
    private String id;
    private String companyName;
    private String companyAddress;
    private String companyCity;
    private String companyStateOrProvince;
    private String companySize;
    private String companyCode;
    private String companyEmail;
    private String companyPhone;
    private String natureOfBusiness;
    private String password;

    // Static method to generate 20 sample companies
    public static List<Company> generateSampleCompanies() {
        List<Company> companies = new ArrayList<>();
        String[] businessTypes = {"Construction", "IT Services", "Agriculture", "Healthcare", "Energy"};

        for (int i = 1; i <= 20; i++) {
            String businessType = businessTypes[i % businessTypes.length]; // Cycle through business types
            companies.add(new Company(
                    UUID.randomUUID().toString(),  // Unique ID
                    "Company " + i,
                    i + " Main Street",
                    "Harare",
                    "Harare",
                    i % 2 == 0 ? "Large" : "Small",  // Alternate between Large & Small
                    "COMP" + i,  // Unique company code
                    "company" + i + "@test.com",
                    "+263 77" + (1000000 + i),  // Generate phone numbers
                    businessType,
                    "password" + i  // Sample passwords
            ));
        }
        return companies;
    }

}

File: frontend/src/types/company.ts
export interface Company {
  id: string;
  email: string;
  companyName: string;
  location: string;
  registrationNumber: string;
  field: string;
  companySize: string;
}

export interface LoginCredentials {
  email: string;
  password: string;
}

export interface AuthResponse {
  token: string;
  company: Company;
}

export interface RegisterFormData {
  companyEmail: string;
  password: string;
  companyName: string;
  companyCity: string;
  companyStateOrProvince: string;
  companyAddress: string;
  companyCode: string;
  companyPhone: string;
  natureOfBusiness: string;
  companySize: string;
}
